### ISSUE TITLE
Focal Port game crash...

### ISSUE THREAD

**Issue by MAGGen-hub** [SamsTheNerd/ducky-periphs#12]
Well I tried hex casting mod... And that happend after I inserted the Focus into Focal Port...
```
> // Oops.
> 
> Time: 2023-04-15 21:00:07
> Description: Unexpected error
> 
> java.lang.NullPointerException: Cannot invoke "net.minecraft.class_2487.method_10573(String, int)" because "tag" is null
> 	at at.petrak.hexcasting.common.lib.hex.HexIotaTypes.getTypeFromTag(HexIotaTypes.java:95)
> 	at at.petrak.hexcasting.common.lib.hex.HexIotaTypes.getDisplay(HexIotaTypes.java:142)
> 	at com.samsthenerd.duckyperiphs.hexcasting.DuckyCastingClient.lambda$setupScryingDisplayers$2(DuckyCastingClient.java:62)
> 	at at.petrak.hexcasting.api.client.ScryingLensOverlayRegistry.getLines(ScryingLensOverlayRegistry.java:157)
> 	at at.petrak.hexcasting.client.HexAdditionalRenderers.tryRenderScryingLensOverlay(HexAdditionalRenderers.java:167)
> 	at at.petrak.hexcasting.client.HexAdditionalRenderers.overlayGui(HexAdditionalRenderers.java:45)
> 	at net.fabricmc.fabric.api.client.rendering.v1.HudRenderCallback.lambda$static$0(HudRenderCallback.java:27)
> 	at net.minecraft.class_329.handler$dfm000$fabric-rendering-v1$render(class_329.java:4996)
> 	at net.minecraft.class_329.method_1753(class_329.java:393)
> 	at net.minecraft.class_757.method_3192(class_757.java:862)
> 	at net.minecraft.class_310.method_1523(class_310.java:1177)
> 	at net.minecraft.class_310.method_1514(class_310.java:768)
> 	at net.minecraft.client.main.Main.method_44604(Main.java:244)
> 	at net.minecraft.client.main.Main.main(Main.java:51)
> 	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:462)
> 	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
> 	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
> 
```

Can't enter the world now...

**Comment by SamsTheNerd** [comment]
what was on the focus? I'm guessing just empty?

**Comment by MAGGen-hub** [comment]
> what was on the focus? I'm guessing just empty?

You right. It was empty (just crafted) before I inserted it... So guess it cased the crash...

**Comment by MAGGen-hub** [comment]
Maybe `not null` check for `tag` will be enough to fix the problem?

**Comment by MAGGen-hub** [comment]
So... Am I right: to fix it right now I need reenable the mod to destroy focal port block, then create new one and create focus, then make it not empty, and only then insert into port. Right?

**Comment by SamsTheNerd** [comment]
yeah, it should be an easy fix when I can get to it. The issue is specifically when you use a scrying lens it tells you info on the focus and that's where I forgot to add a null check. 

And yes, you'll need to get rid of the focal port or somehow remove your scrying lens from your face/hand. You can have an empty focus in there as long as you don't have the scrying lens equipped, but yeah for now it's probably safer to just write null to the focus before putting it in there.

**Comment by MAGGen-hub** [comment]
> yeah, it should be an easy fix when I can get to it. The issue is specifically when you use a scrying lens it tells you info on the focus and that's where I forgot to add a null check.
> 
> And yes, you'll need to get rid of the focal port or somehow remove your scrying lens from your face/hand. You can have an empty focus in there as long as you don't have the scrying lens equipped, but yeah for now it's probably safer to just write null to the focus before putting it in there.

Oh... The issue in the lens... Pff... I still don't understand why I should use it (it do not display any important stuff for me [at last for now])... Okay, then thanks for help. Oh! Another question: do you see any addons or packs that combine this lens with engineer glasses from create mod (could be cool to use both tools at the same time)?

**Comment by MAGGen-hub** [comment]
Oh... Another bug... I will open new issue...

**Comment by SamsTheNerd** [comment]
i have not seen any addons with that, i've heard people suggest it in the discord but i don't know of anything that adds it in. I know in a future update it's supposed to be able to extend your casting grid but idk when that's coming.

**Comment by SamsTheNerd** [comment]
should be fixed in v1.2.0
